src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs

```diff
@@ -186,53 +186,56 @@ public static (SourceText text, TextSpan semicolonSpan) ConvertNamespaceDeclarat
-            }
+            var originalHadTrailingNewLine = namespaceDeclaration.GetTrailingTrivia() is [.., { RawKind: (int)SyntaxKind.EndOfLineTrivia }];
 
-            // Otherwise, ensure there's a blank line between the namespace line and the first body member. Don't bother
-            // with this though if we already separated things by moving a pp directive (like a #else) from the open brace
-            // to the first token.
-            if (firstBodyToken.Kind() != SyntaxKind.EndOfFileToken &&
-                !HasLeadingBlankLine(firstBodyToken, out _) &&
-                !namespaceDeclaration.OpenBraceToken.LeadingTrivia.Any(t => t.IsDirective))
+            // now, intelligently trim excess newlines to try to match what the original namespace looked like.
+            while (fileScopedNamespace.HasTrailingTrivia)
             {
-                fileScopedNamespace = fileScopedNamespace.ReplaceToken(
-                    firstBodyToken,
-                    firstBodyToken.WithPrependedLeadingTrivia(SyntaxFactory.CarriageReturnLineFeed));
-            }
+                var trailingTrivia = fileScopedNamespace.GetTrailingTrivia();
 
-            // Copy leading trivia from the close brace to the end of the file scoped namespace (which means after all of the members)
-            fileScopedNamespace = fileScopedNamespace.WithAppendedTrailingTrivia(namespaceDeclaration.CloseBraceToken.LeadingTrivia);
+                // if the new namespace doesn't end with a newline, nothing for us to do.
+                if (trailingTrivia is not [.., { RawKind: (int)SyntaxKind.EndOfLineTrivia }])
+                    break;
 
-            // If the previous namespace declaration had no trailing trivia and the last member only has a newline, then assume the user
-            // doesn't want that newline any more.
-            if (!namespaceDeclaration.HasTrailingTrivia &&
-                namespaceDeclaration.CloseBraceToken.GetPreviousToken() is var lastMemberToken &&
-                lastMemberToken.TrailingTrivia is [{ RawKind: (int)SyntaxKind.EndOfLineTrivia }])
-            {
-                fileScopedNamespace = fileScopedNamespace.WithoutTrailingTrivia();
+                // if the original had a newline, then we only want to trim the newlines as long as there is still one
+                // left at the end.
+
+                if (originalHadTrailingNewLine && trailingTrivia is not [
+                    ..,
+                    { RawKind: (int)SyntaxKind.EndOfLineTrivia or (int)SyntaxKind.EndIfDirectiveTrivia or (int)SyntaxKind.EndRegionDirectiveTrivia },
+                    { RawKind: (int)SyntaxKind.EndOfLineTrivia } ])
+                {
+                    break;
+                }
+
+                // New namespace has excess newlines, remove the last one and try again.
+                fileScopedNamespace = fileScopedNamespace.WithTrailingTrivia(
+                    trailingTrivia.Take(trailingTrivia.Count - 1));
             }
```

Comment:
new logic is different, but seems to fit better with the before code.  First, we continuously trim down any multiple-new-lines we have.  We also remove the final new-line if hte original code didn't end with a newline either.